### ISSUE TITLE
Remove packages from GHCR.io for PRs which were merged

### ISSUE THREAD

**Issue by grzanka** [yaptide/app#803]
Right now the retention policy works for PRs which were closed, like this one:
https://github.com/yaptide/yaptide/pull/799

It is not triggered for PRs which were merged, like this one:
https://github.com/yaptide/yaptide/pull/802

See:
![image](https://github.com/yaptide/yaptide/assets/7374733/93ed4d50-fb70-450f-845f-65970e61b993)

**Comment by macsko** [comment]
I think it's not that straightforward. We can see on the screenshot that it worked for #798 and #801, which were merged. I think we need to gather more data to see what caused the issue. Perhaps it was just a problem on GitHub side. I found a few issues that the `on pull_request closed` action is not always tiggering: https://github.com/orgs/community/discussions/26657.

If the issue keeps happening I will try to switch from `pull_request` to `pull_request_target` action trigger. We can also try to cleanup all PR packages that are older than e.g. 2 months even if they are the newest in the PR.
